server/bundles/io.cloudbeaver.model/src/io/cloudbeaver/model/rm/local/LocalResourceController.java

```diff
@@ -483,14 +487,70 @@ public String moveResource(
         } catch (IOException e) {
             throw new DBException("Error moving resource '" + oldResourcePath + "'", e);
         }
+
+        log.debug("Moving resource properties");
         // Move properties
-        getProjectMetadata(projectId, false).moveResourceProperties(oldResourcePath, newResourcePath);
+        final var project = getProjectMetadata(projectId, false);
+        project.moveResourceProperties(oldResourcePath, newResourcePath);
+
+        log.debug("Moving children properties");
+        try {
+            moveChildrenProperties(project, projectId, newTargetPath, oldTargetPath.getFileName());
+        } catch (IOException | DBException e) {
+            throw new DBException("Unable to move child resource properties", e);
+        }
 
         fireRmResourceDeleteEvent(projectId, rmOldResourcePath);
         fireRmResourceAddEvent(projectId, newResourcePath);
         return DEFAULT_CHANGE_ID;
     }
 
+    private void moveChildrenProperties(
+            @NotNull final DBPProject project,
+            @NotNull final String projectId,
+            @NotNull final Path actualParentPath,
+            @NotNull final Path oldParentName
+    ) throws IOException, DBException {
+        final var projectPath = getProjectPath(projectId);
```

Comment:
should we check that actualParentPath is folder or does it automatically checked in walkFileTree?

Reply:
Yes, when we pass a file to `Files#walkFileTree` it calls `FileVisitor#visitFile`